verafelix/reprograma.py

```diff
@@ -0,0 +1,202 @@
+        "Notas": [9.0, 9.0, 8.5],
+        "Presença": [True, True, True, True, True],
+        "Participação": 8.8
+    },
+    ("Beatriz", "Lima"): {
+        "Turma": "Turma C",
+        "Notas": [8.0, 7.5, 7.0],
+        "Presença": [True, True, True, False, True],
+        "Participação": 7.5
+    }
+}
+
+
+def main():
+    print("\n---  Seja bem vinda a Escola do Reprograma!  ---")
+    print("Sistema de informações de alunas")
+
+    while True:
+        cod_opcao = obter_opcao()  # Obtém a opção escolhida pelo usuário
+
+        if cod_opcao == 1:
+            incluir_nova_aluna()  # Chama a função para incluir nova aluna
+        elif cod_opcao == 2:
+            consultar_lista_alunas()  # Chama a função para consultar lista de alunas
+        elif cod_opcao == 3:
+            consultar_faltas_aluna()  # Chama a função para consultar faltas de uma aluna
+        elif cod_opcao == 4:
+            consultar_notas_aluna()  # Chama a função para consultar notas de uma aluna
+        elif cod_opcao == 5:
+            consultar_status_aprovacao()  # Chama a função para consultar status de aprovação
+        elif cod_opcao == 6:
+            print("Encerrando o programa...")
+            break  # Encerra o programa
+
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            # Solicita ao usuário para escolher uma opção
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                     "1 - Incluir uma nova aluna\n"
+                                     "2 - Consultar lista de alunas\n"
+                                     "3 - Consultar faltas da aluna\n"
+                                     "4 - Consultar notas da aluna\n"
+                                     "5 - Consultar status de aprovação\n"
+                                     "6 - Sair do sistema\n"
+                                     "Opção: "))
+
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 6).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+
+    return codigo_opcao
+
+
+def incluir_nova_aluna():
+    # Solicita os dados da nova aluna
+    nome = input("Digite o nome da aluna: ")
+    sobrenome = input("Digite o sobrenome da aluna: ")
+    turma = input("Digite a turma da aluna: ")
+    notas = list(
+        map(float, input("Digite as notas da aluna separadas por espaço: ").split()))
+    presenca = list(map(lambda x: x.lower() == 'true', input(
+        "Digite as presenças da aluna (True/False) separadas por espaço: ").split()))
+    participacao = float(input("Digite a nota de participação da aluna: "))
+
+    # Valida que todas as notas estão no intervalo permitido
+    if not all(0 <= nota <= 10 for nota in notas):
+        print("Erro: Todas as notas devem estar entre 0 e 10.")
+        return
```

Comment:
Já que fez uma adição de validação para a lista de notas, poderia acrescentar uma valiação para a nota de participação também :)

**Sugestão:**
```suggestion
    if any(nota < 0 or nota > 10 for nota in notas):
        print("Erro: As notas devem estar entre 0 e 10.")
        return

    if participacao < 0 or participacao > 10:
        print("Erro: A nota de participação deve estar entre 0 e 10.")
        return
```